lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -2336,6 +2336,305 @@ class ConvertCastEquivalentOp : public OpConversionPattern<OpTy> {
+    Value innerDim_1b =
+        rewriter.create<arith::SubIOp>(loc, innerDim_1a, oneIndex);
+    Type int64type = mlir::IntegerType::get(op->getContext(), 64);
+    Value innerDim_0c =
+        rewriter.create<arith::IndexCastOp>(loc, int64type, innerDim_0b);
+    Value innerDim_1c =
+        rewriter.create<arith::IndexCastOp>(loc, int64type, innerDim_1b);
+    Type floatType = mlir::FloatType::getF32(op->getContext());
+    Value innerDim_0d =
+        rewriter.create<arith::SIToFPOp>(loc, floatType, innerDim_0c);
+    Value innerDim_1d =
+        rewriter.create<arith::SIToFPOp>(loc, floatType, innerDim_1c);
+    Value two = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getFloatAttr(floatType, 2.0));
+    Value innerDim_0e = rewriter.create<arith::DivFOp>(loc, innerDim_0d, two);
+    Value innerDim_1e = rewriter.create<arith::DivFOp>(loc, innerDim_1d, two);
+
+    // calculate grid dimensions
+    Value grid = adaptor.getGrid();
+    auto gridType = grid.getType().cast<RankedTensorType>();
+    auto gridElementType = gridType.getElementType();
+    SmallVector<int64_t> gridShape =
+        makeShapeTorchCompatible(gridType.getShape());
+    auto gridRank = gridType.getRank();
+    Value zeroIndex = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    SmallVector<Value> extractGridOffsets_0(gridRank, zeroIndex);
+    SmallVector<Value> extractGridShape = getTensorSizes(rewriter, loc, grid);
+    SmallVector<Value> extractGridStride(gridRank, oneIndex);
+    int64_t lastGridDim = gridRank - 1;
+    Value twoIndex = rewriter.create<arith::ConstantIndexOp>(loc, 2);
+    extractGridShape[lastGridDim] = oneIndex;
+    extractGridStride[lastGridDim] = twoIndex;
+    SmallVector<Value> extractGridOffsets_1(gridRank, zeroIndex);
+    extractGridOffsets_1[lastGridDim] = oneIndex;
+
+    auto grid_0 = rewriter.create<tensor::ExtractSliceOp>(
+        loc, grid, extractGridOffsets_0, extractGridShape, extractGridStride);
+    auto grid_1 = rewriter.create<tensor::ExtractSliceOp>(
+        loc, grid, extractGridOffsets_1, extractGridShape, extractGridStride);
+    SmallVector<int64_t> gridShapeExtracted =
+        makeShapeTorchCompatible(grid_0.getType().getShape());
+    Value gridResultEmpty_0 = rewriter.create<tensor::EmptyOp>(
+        loc, gridShapeExtracted, gridElementType);
+    Value gridResultEmpty_1 = rewriter.create<tensor::EmptyOp>(
+        loc, gridShapeExtracted, gridElementType);
+    Value lowerEmpty_0 =
+        rewriter.create<tensor::EmptyOp>(loc, gridShapeExtracted, int64type);
+    Value lowerEmpty_1 =
+        rewriter.create<tensor::EmptyOp>(loc, gridShapeExtracted, int64type);
+    Value upperEmpty_0 =
+        rewriter.create<tensor::EmptyOp>(loc, gridShapeExtracted, int64type);
+    Value upperEmpty_1 =
+        rewriter.create<tensor::EmptyOp>(loc, gridShapeExtracted, int64type);
+    Type int1type = mlir::IntegerType::get(op->getContext(), 1);
+    Value notValidEmpty_0 =
```

Comment:
You won't need to materialize the not valid tensor. You can just check the produced values.

Reply:
If a coordinate is not valid how do I check if the produced value is valid or not? 
// I guess you are saying : 
a = inp[x,y]
if a is not valid 
    a = 0

Reply:
combined this within one larger linalg::genericop